source/code/plugins/oms_configuration.rb

```diff
@@ -173,8 +173,11 @@ def get_azure_resid_from_hybridagent()
               OMS::Log.warn_once("Failed to read the IMDS Endpoint from #{@@HybridRPConfFile}. Failed to fetch azure resource id")
               return nil
             end
-            azureHybridIMDSEndpoint = azureHybridIMDSEndpoint.split("=")[1][0..-2]
+
+            #Subbing localhost with 127.0.0.1 because Ruby's resolv-replace module (imported in oms_common.rb fails to correctly resolve localhost)
+            azureHybridIMDSEndpoint = azureHybridIMDSEndpoint.split("=")[1][0..-2].sub("localhost","127.0.0.1")
```

Comment:
why do you need `[0..-2]`

Reply:
It is to remove the '**\"**' at the end of the url. the scan() function above adds the extra slash in between the url and the end quotes.